### PR TITLE
ssh/tailssh: set default Tailscale SSH $PATH for non-interactive commands

### LINKED ISSUE #5285: Tailscale ssh lacks $PATH for non-interactive sessions (VSCode, Xpra, etc)
### What is the issue?

When I connect to a remote machine using VSCode Remote SSH, my $PATH gets stomped with a VSCode path which is supposed to be appended, not overwrite my entire path.

This may be a VSCode bug ([here](https://github.com/microsoft/vscode-remote-release/issues/6975)) but I'm posting here because there's definitely some interaction with tailscale. This _only_ occurs via tailscale ssh.

Works:
- SSH in terminal not via tailscale
- SSH in terminal via tailscale (tailscale up --ssh)
- SSH via VSCode Remote SSH not via tailscale

Stomps $PATH:
- SSH via VSCode Remote SSH via tailscale

### Steps to reproduce

- `tailscale up --ssh`
- In VSCode, connect to that remote host (disconnect and reconnect several times)

### Are there any recent changes that introduced the issue?

_No response_

### OS

Linux

### OS version

Ubuntu 20.04

### Tailscale version

1.28.0

### Bug report

_No response_

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes #5285

Co-authored-by: @andrew-d

### CODE REVIEW

**Review by andrew-d — APPROVED** [review]
LGTM; we should probably write tests for this if we make it more complicated in the future, but this is easy enough for now.

File: ssh/tailssh/tailssh_test.go [hunk 1]
```diff
@@ -755,3 +755,27 @@ func TestAcceptEnvPair(t *testing.T) {
...
+			u, ""},
+	}
+
+	for i, tt := range tests {
+		got := pathFromPAMEnvLine([]byte(tt.line), tt.u)
+		if got != tt.want {
+			t.Errorf("%d. got %q; want %q", i, got, tt.want)
+		}
+	}
+
```
Comment (andrew-d) [inline ssh/tailssh/tailssh_test.go:780]: Extra newline

File: ssh/tailssh/tailssh_test.go [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build linux || darwin
+//go:build linux || darwin || freebsd
```
Comment (andrew-d) [inline ssh/tailssh/tailssh_test.go:5]: ???

File: ssh/tailssh/incubator.go [hunk 3]
```diff
@@ -567,7 +579,60 @@ func envForUser(u *user.User) []string {
...
+func pathFromPAMEnvLine(line []byte, u *user.User) (path string) {
+	if !mem.HasPrefix(mem.B(line), mem.S("PATH")) {
+		return ""
+	}
+	rest := strings.TrimSpace(strings.TrimPrefix(string(line), "PATH"))
+	if quoted, ok := strs.CutPrefix(rest, "DEFAULT="); ok {
+		if path, err := strconv.Unquote(quoted); err == nil {
+			path = strings.NewReplacer(
+				"@{HOME}", u.HomeDir,
+				"@{PAM_USER}", u.Username,
```
Comment (andrew-d) [inline ssh/tailssh/incubator.go:627]: *Technically* these can be escaped; from [here](https://man7.org/linux/man-pages/man5/pam_env.conf.5.html):

> Both the $ and @ characters can be backslash escaped to be used as literal values values

Probably fine to ignore for now, though.

**Review by andrew-d — APPROVED** [review]

File: ssh/tailssh/incubator.go (hunk 3 above)
Comment (bradfitz) [reply ssh/tailssh/incubator.go:631]: Happy to ignore :)

File: ssh/tailssh/tailssh_test.go (hunk 2 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh_test.go:5]: Oh, Tailscale SSH FreeBSD support was added in Tailscale 1.34 so figured it should start running tests too. I'll remove for now and add in its own commit.

File: ssh/tailssh/tailssh_test.go (hunk 1 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh_test.go:780]: done

File: ssh/tailssh/tailssh_test.go [hunk 4]
```diff
@@ -755,3 +756,43 @@ func TestAcceptEnvPair(t *testing.T) {
...
+	}
+	for i, tt := range tests {
+		got := pathFromPAMEnvLine([]byte(tt.line), tt.u)
+		if got != tt.want {
+			t.Errorf("%d. got %q; want %q", i, got, tt.want)
+		}
+	}
+}
+
+func TestPathFromPAMEnvLineOnNixOS(t *testing.T) {
```
Comment (bradfitz) [inline ssh/tailssh/tailssh_test.go:781]: added this new test too to give us an early warning if NixOS changes

Comment (andrew-d) [reply ssh/tailssh/tailssh_test.go:781]: Nice; I like it!

**Pull request merged**